### DISCUSSION TITLE
Moving forum to here?  Why?

### DISCUSSION THREAD

**Discussion by Jibun-no-Kage** [micropython/micropython#9106] [General]
Moving forum to here?  Why?  Just asking, wondering why this is being done?  Have no problem using GitHub, but others may not be as comfortable with it in a typical forum context?

**Comment by jimmo** [comment]
See background at https://github.com/micropython/micropython/issues/8795 and https://forum.micropython.org/viewtopic.php?f=8&t=12577, as well as today's announcement here https://forum.micropython.org/viewtopic.php?f=8&t=12971

Other than the email filtering (which I seem to have a workaround for), I am very impressed with Discussions so far.

**Comment by bulletmark** [comment]
I visit the forum frequently on micropython.org and the first thing I press is the "New posts" button. How do you get similar functionality here? I find it awkward to find new posts here that I have not seen.

Very much prefer posting in Github flavored markdown so overall I do prefer it here.

**Reply by jimmo** [reply]
That was exactly how I used the old forum too, but it always frustrated me that the updated threads would then disappear from the New Posts lists after some time (~hours) so if I didn't reply the first time I saw them then they'd disappear.

On GitHub, my workflow is now that I have email subscription set up to the micropython repository, so now I get an email feed of updates, which is actually perfect and suits my workflow exactly. (I realise that my use case is not at all the typical forum user though).

To answer your question though, my experience so far has been that the default view in GH Discussions ("New" rather than "Top") actually matches exactly the "New posts" behavior of phpBB... i.e. you see the threads, sorted by which one had the most recent reply. *Except* (and I guess this is your point) it doesn't remove posts that you've already seen. Sounds like you want to see a "show me updates since I was last here" view? For me, because I'm using email as the primary way to see new activity, that's how I get the "mark as read" behaviour.

FWIW, there's some logic that causes posts in the list to be bolded, which appears to be somewhat correlated with "unread" but I'm not clear exactly how it works. (Edit: I think my confusion here is that if _I_ reply to a post, it will still consider that unread and therefore bold until I open the thread again... That seems like a bug).

(Side note: There were some suggestions in the discussion about moving about using the RSS feeds from phpBB (@cwalther) which worked well for the use case of seeing new threads, but didn't help for getting notified about replies to existing threads)

**Reply by cwalther** [reply]
> There were some suggestions in the discussion about moving about using the RSS feeds from phpBB 

For what it’s worth, I have now switched to exactly the same e-mail workflow as you, and so far it seems like for my purposes it’s just as good as the RSS feeds. It gives me my local offline searchable archive and read/unread state under my control.

> but didn't help for getting notified about replies to existing threads

That’s not true, there are feeds containing all posts (I only use those), not just feeds containing threads.

**Reply by jimmo** [reply]
> That’s not true, there are feeds containing all posts (I only use those), not just feeds containing threads.

OK thank you @cwalther . I didn't see the posts (not topics) feed at the time. TBH I was really confused by the way the RSS stuff worked in phpBB, and in particular why all the feed links seemed to be commented out in the micropython.org forum HTML.

Ahh, just managed to find it (I see how I missed it, silly mistake) but it's not a very good experience because my RSS reader cannot thread them (can any RSS readers do that?). The GMail experience is heaps better.

**Reply by cwalther** [reply]
Right, the threading is a problem in my feed reader too. A makeshift way is sorting by subject, although that breaks down when people change the subject mid-thread (as @peterhinch often does).

**Reply by cwalther** [reply]
And I have now discovered a gotcha of the e-mail workflow: You need to enable _Includes your own updates_ in [Settings ▶︎ Notifications ▶︎ Customize email updates](https://github.com/settings/notifications), otherwise you don’t get e-mail notifications of your own posts, and then these are missing from your archive. (Even though they’re obviously the most important ones! 😉)

**Comment by Jibun-no-Kage** [comment]
I have to say, I don't like the GitHub migration of the forum.  I use many forums, and many of them used the same environment, which was of great benefit.  I can and will use GitHub discussions, since I am already familiar with it, so there is no culture shock issue on my account, but if we retained or moved back to the original forum, I would appreciate it.  I understand why the move was done, and I can see the advantage, but that said, I just liked the old forum better.

**Reply by jimmo** [reply]
> I understand why the move was done, and I can see the advantage, but that said, I just liked the old forum better.

Could you give some specific reasons?

**Comment by Jibun-no-Kage** [comment]
Sure....  But let me be clear, I have used GitHub for years now, event though I find other environments easier to use.  So in point of fact, at some level I have been complicit in not encouraging changes to GitHub look and feel, the ease of use.  And, of course this is all subjective at some level.  But since you asked, you have been warned. :)

To pick one concrete example, the ease of how code or command-line out snippets, can be presented or pasted is one key issue.  I have tried 100s of times to get code snippets, or console output results to display right in GitHub, without pain, for years.  It is a PITA, and has been for some time.  Yes, I know it is supposed to be easy in GitHub.... I just want to paste and be done, not worry if I am in Windows, or Linux, or my phone, or using browser A or B,, not have to count tick marks, to spaces to indent, when select the type (link or code or whatever), then paste, done.  The idea that GitHub wants to parse text its own way, is just dumb.  One of my favorites... '#' as bold header tag?  Which means anything you past from a Linux root level terminal session is now 24 point bold (header) in preview mode or after you save your comment?  Really?  This has been discussed 1000s of times (cough), but GitHub never addresses this issue,. to make it as 'easy' as even most forums that (like the original MP forum), or Node Red use, even the Raspberry Pi form pasting links, code, etc., is easier than GitHub.  Any interface should be intuitive, painless to use,  Or why bother... but GitHub just goes on its own way.

Only read if interested...

History illustrates this, the world of DOS learned this the hard way when the Macintosh OS was first released.  Which was based on XPARC concepts that had been around for years.  And soon after Macintosh OS was seen by others, as a game changer, for Windows 1.0 to appear because it was a matter of survival, but until it did many programmers that refused to accept a thing called a GUI, often said, a GUI a pretty fad, or a cute idea.  They just did not want to make the investment in time and resources, to change, I know this first hand, having worked supporting PC/DOS, DEC VAX, oriented developers and then later for Apple during my University days.  I remember the discussions with friends over this specific debated subject.

Present day, why has GitHub not been forced to improve their interface to be easier and smoother to use?  Because the majority of GitHub is NOT users... BUT programmers, which have a different mind-set and tolerance for such.  I suggest that if the majority of people using GitHub were in fact typical USERS, GitHub would have changed long ago.  Programmers are NOT users, no matter what argument presented, they are two different mind-sets... and anyone that can't see this point, NEVER will.  But I can definitely tell you this is true.

I have lived it, I have even rejected solution designs for Fortune 10 firms, as an early adapter, emerging technologist, worth multi million dollar proposals, over ease-of-use issues.  One vendor which I will not name, but their name starts with a 'D' and is a 4 letter word, was not happy with me over ease-of-use more than once, I said "As an engineer, I pulling on my end-user hat...", the vendor replied, "Shit... I HATE when he does that...", to which my boss replied, to everyone in the meeting, "Here we go... Again.., The operations team is going to love this."  Everyone laughed, the vendor later thanked me for pointing out some design issues that they had not seen as yet, or any other customer had commented on.  My boss took me out to dinner, and when I asked, so I stop pushing back on design issues, he replied, and I quote, "Hell no, you are one of few engineers that came from operations, so you know when something is going to be painful to use day in and day out, as a user."

The bottom line, the vendor improved their product, and I was told that other customers told the vendor their ease of use was great... I WONDER WHERE THEY GOT THAT IDEA FROM?  :)

**Comment by bulletmark** [comment]
@Jibun-no-Kage your only specific complaint was how hard it is to paste code. I'ts amazing how people can have such different opinions because for me that would actually be the most compelling advantage of github compared to the old forums. Code is **much** easier to paste (between ``` markers) and offers syntax highlighting as well:

```python
#!/usr/bin/python
from pathlib import Path

Path('afile').write_text('Look how easy it is and doesn't this look better?!')
```

**Comment by Jibun-no-Kage** [comment]
No, I with respect, disagree.  I only provided one example, for the sake of the discussion, I also illustrated a general concept, that in total, GitHub is not as easy to use as the previous forum, and is quite different from many of the forums I use, that are identical to the legacy MP forum.  As I noted in my reply, if you don't see my point, you never will.  Programmers don't understand end-user issues, well, they never have as a group, this why we have to have business and systems analysts interact with users and then in turn communicate to developers/programmers.  Why major universities make all Computer Science majors take one or more classes on systems analysis.

There is an entire job classification to address a 'classic' issue that has been part of the industry since it origin.  I say this, based on experience of 30+ years in the industry.  Your response suggests, you are focusing on the specific example, not the total problem, IMHO.  My specific experience coming from operations, end-user perspective, not as a typical engineer, so yes I have a defined bias, one that helped my career extensively, which I also qualified and illustrated.

Once I was an engineer, an enterprise architect designing virtual infrastructure, I realized that even I, had begun seeing the world as an engineer, it was a surprise to be sure, once I realized this.  And, even I, as much as GitHub has frustrated me over time, I accepted it, and did not, as maybe I should have, tried to influence the design/function of its end user interface.

One of my best friends, was a developer/programmer for decades, then became a business analyst, and has stated on more than one occasion, how frustrating it is, working with developers that refuse to see his perspective, he has voiced more than once... was I ever this pig headed about my code design?  I reply is always.... I don't know, I only met you have you ceased being a jerk!

**Reply by jimmo** [reply]
An bit of context that is important to remember is that there are three categories of contributors to this forum:
 - People who have never used any type of markup/down.
 - People who come from existing forums and could have used any one of many different schemes.
 - People who come from any of the many places that have converged on Markdown (e.g. GitHub, Slack, Discord, and many more, including other popular forum software like Discourse).

I understand that you feel very strongly about your previous experience with phpBB (or other), but it's not representative of the entire range of contributors we see here. For what it's worth, I regularly (i.e. at times, daily) saw posts on the old forum where the formatting of code blocks was completely wrong (or not attempted), and as a moderator have done behind-the-scenes edits countless times.

It's worth mentioning as well that one really useful feature of Markdown formatting that is completely unsupported by phpBB is inline code formatting, which I use in almost every post to refer to variable or function names.

Anyway, I have updated the [Welcome post](https://github.com/orgs/micropython/discussions/9111) with clearer instructions.

**Reply by bulletmark** [reply]
> It's worth mentioning as well that one really useful feature of Markdown formatting that is completely unsupported by phpBB is inline code formatting, which I use in almost every post to refer to variable or function names.

Same. Actually, if you look over my old posts on MicroPython.org (and other similar forums) you may notice I have used back-ticks there for inline code for a while. Even though their content was not rendered, I did that assuming many readers would understand that the back-ticks delineated inline code. It was also a very subtle "protest" that the forum software was lacking the feature :-) 

So my summary of the move here is that within a thread, Github discussions is **much** better. However it is deficient at the thread overview level as you can't easy see or navigate to threads updated since your last visit, has poor grouping, and has very poor use of screen real-estate (at least on my large screen PC). I guess it is likely that Github will improve these issues though.

**Comment by Jibun-no-Kage** [comment]
I think every environment has its quirks.  This is why I made the point that, at least for me, there is no culture shock issue in using GitHub from this point on for MP discussions.  The requested was made that I explain or illustrate, I believe I have, or did.  As for how anyone interprets or receives such communication is up to them.  As for me, every time I do a cut or paste that formats funny, I will dealt with it.  But don't be surprised if your ears ring or you get a cold chill, because my impolite utterances, likely in German, are received across the vastness of the transcendental ether on some ultrasonic wavelength humans rarely are attuned to.  LOL

**Reply by scruss** [reply]
There is one major problem with Github Discussions that I don't see any way out of: you can't permanently unsubscribe to a thread. You can unsubscribe you, but if someone @-mentions you, you're back subscribed again. This mechanism has potential for abuse. When I unsubscribe, I need the platform to Do What I Mean.

For instance, I contributed to this thread “[Micropython ADC Noisier than Arduino on ESP32](https://github.com/orgs/micropython/discussions/9227)”, and then unsubscribed because I couldn't contribute anything more useful. I got mentioned, and *bam!*, I'm resubscribed again. Not what I want at all.

**Comment by peterhinch** [comment]
I find the effect on notifications difficult. Before the change I would scan notifications for any issues or PR's of interest, then go to the forum. Now notifications is a huge list of issues, PR's and discussion topics all jumbled together.

Is there a way to prevent discussions from populating notifications?

**Reply by jimmo** [reply]
You can create custom views (filters) for the notifications page. I have an "MP Issues & PRs" one that should by the sounds of it be equivalent to what you used to see in the full notifications list. Then a "MP Discussions" that should be the same as "New Posts" from the phpBB forum.

![Screenshot_2022-09-08_22-31-34](https://user-images.githubusercontent.com/5102768/189122407-aadaf6be-c15f-4cb0-ac1f-c35809f26e00.png)

My workflow is entirely by email though with the two filters I posted in the Welcome message (and this works _really_ well now with discussions as I just have the two labels and mark-as-unread etc and can do this from my phone too). I use the GH notifications in the web view (or their mobile app) although it has improved considerably. I just much prefer managing things through email because it's much more responsive and can get through the items faster.

FWIW "Mark as unread" is a huge feature for me compared to phpBB, and is much preferable to the giant collections of tabs I kept open to keep track of things in the old forum.  (again I'm probably not the typical user)

**Reply by peterhinch** [reply]
That works nicely, thank you. Will issues and PR's raised against repos I am watching show up? If not, is there a way to create a filter that ensures I get to see them?

**Reply by jimmo** [reply]
repos you're watching... hrmm. I guess you could create a filter which is `is:issue-or-pull-request` and at least that will get pretty much what your inbox was prior to Discussions. It's interesting there's no `reason:watching`.

Another useful filter would be `org:peterhinch` I guess.
